### PR TITLE
fix(file-router): ignore "config" export for HMR

### PR DESCRIPTION
**Pull request by Lodin**
Allows correct HMR update in case `export const config` exists in the route file.

### CODE REVIEW

File: packages/ts/hilla-file-router/src/vite-plugin.ts [hunk 1]
```diff
@@ -87,5 +90,20 @@ export default function vitePluginFileSystemRouter({
       server.watcher.on('change', changeListener);
       server.watcher.on('unlink', changeListener);
     },
+    transform(code, id): Promise<TransformResult> | TransformResult {
+      if (id.startsWith(fileURLToPath(_viewsDir))) {
```
Comment (Legioth) [inline packages/ts/hilla-file-router/src/vite-plugin.ts:94]: Should maybe also ignore files starting with `_` inside the views directory?

Comment (Lodin) [reply packages/ts/hilla-file-router/src/vite-plugin.ts:94]: Right, good point

**Comment by taefi** [comment]
The java test failure would be alleviated after #2130 is merged, stay tuned...

File: packages/ts/hilla-file-router/src/vite-plugin.ts (hunk 1 above)
Comment (Lodin) [reply packages/ts/hilla-file-router/src/vite-plugin.ts]: Done

**Review by Legioth — COMMENTED** [review]
I don't pretend to have a good enough understanding of how HMR works without some additional explanations but I might still consider this change to be good enough if it's guarded by the integration test that we discussed.

**Comment by taefi** [comment]
> I don't pretend to have a good enough understanding of how HMR works without some additional explanations but I might still consider this change to be good enough if it's guarded by the integration test that we discussed.

Basically, I was struggling with the fact that I really don't know how HMR is working and how that regex is helping there :)

**Comment by Legioth** [comment]
This is basically one of those cases where it would be helpful with some comments in the code to give some additional contest in the form of an examples of what the targeted code looks before the transformation and what it is supposed to look like after.

**Review by platosha — APPROVED** [review]

**Review by platosha — APPROVED** [review]

**Pull request merged**